packages/react-components/src/components/MessageThread.tsx

```diff
@@ -733,7 +849,14 @@ export const MessageThread = (props: MessageThreadProps): JSX.Element => {
   } = props;
   const onRenderFileDownloads = onRenderFileDownloadsTrampoline(props);
 
-  const [messages, setMessages] = useState<(ChatMessage | SystemMessage | CustomMessage)[]>([]);
+  const [messages, setMessages] = useState<
+    (
+      | ChatMessage
```

Comment:
Add a new type, to merge these all together
type Message = ChatMessage | SystemMessage | CustomMessage | /* @conditional-compile-remove(data-loss-prevention) */ BlockedMessage

(maybe find a better name)